server/pgServer.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const app = express();
+const path = require('path');
+const cors = require('cors');
+const expressStaticGzip = require('express-static-gzip');
+const knex = require('../knexConnection.js');
+const bodyParser = require('body-parser');
+const port = 3005;
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+
+// app.use(express.static('client'));
+app.use('/', expressStaticGzip(path.join(__dirname, '../client'), {
+  enableBrotli: true,
+  orderPreference: ['br', 'gz'],
+  setHeaders: function (res, path) {
+    res.setHeader("Cache-Control", "client, max-age=31536000");
+  }
+}));
+
+app.use(cors());
+
+// ---------------- CRUD FOR POSTGRESQL --------------- //
+
+app.get('/songdata/:id', async (req, res) => {
+  try {
+    var id = req.params.id;
+    if (id > 10000001 || id < 0) {
+      throw new Error(`song_id: ${id} does not exist`);
+    }
+    var song = await knex('songs').where({ song_id: id });
+    res.status(200).json({
+      status: "Success",
+      data: song
+    });
+  } catch (error) {
+    console.log(error)
+  }
+});
+
+app.post('/song', async (req, res) => {
+  // console.log('Request: ', req.body);
+  try {
+    var song = await knex('songs').insert(req.body);
+    res.status(201).json({
+      status: 'Success',
+      data: song
+    });
+  } catch (error) {
+    console.log('Error creating new song: ', error);
+    res.status(400).json({
```

Comment:
Is this more representative of 5xx server errors. on a separate note, you may consider handling bad request explicitly before invoking the DB query?

Reply:
Are you meaning like the error handing I do for the GET to check for a valid input id before initializing the search?

Reply:
Yes.

When trying to insert, there may be other reasons DB may throw an error, that would go to your catch block.

Best practice is 4xx client errors, and 5xx for server errors.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
https://stackoverflow.com/questions/39636795/http-status-code-4xx-vs-5xx